libimage/image.go

```diff
@@ -384,7 +384,7 @@ func (i *Image) removeRecursive(ctx context.Context, rmMap map[string]*RemoveIma
 	// have a closer look at the errors.  On top, image removal should be
 	// tolerant toward corrupted images.
 	handleError := func(err error) error {
-		switch errors.Cause(err) {
+		switch errors.Unwrap(err) {
```

Comment:
This does not the same. errors.Cause(err) is recursive and returns err when it cannot find a cause.
errors.Unwrap() is not recursive and returns nil when it cannot unwrap.

Same for other places where you changed this.